solid/authorlist.rb

```diff
@@ -0,0 +1,68 @@
+require_relative 'validation'
+require_relative '../classes/author'
+require_relative '../serializers/serializearray'
+
+class AuthorList
+  include SerializationArray
+  attr_accessor :list
+
+  def initialize
+    @list = []
+    @input = InputValidation.new
+  end
+
+  def author_attr
+    first_name = @input.input_string('First Name:')
+    last_name = @input.input_string('Last Name:')
+    [first_name, last_name]
+  end
+
+  def add_author
+    first_name, last_name = author_attr
+    author = Author.new(first_name, last_name)
+
+    puts 'Author created successfully'
+    @list << author
+    author
+  end
+
+  def show
+    @list.each do |author|
+      puts "AUTHOR: First Name: #{author.first_name},
+      Last Name: #{author.last_name}, Books/Games: #{author.items.length}"
+    end
+  end
+
+  def select_create_author_for(_item_list)
+    author = select_author
+    return author if author
+
+    author = add_author if @input.input_yes_no('Do you want to create a new author? (y/n):')
+
+    author || Author.new('', '')
+  end
```

Comment:
After having the author, you should add this author in the _item_list using the method add_author